lms/djangoapps/letstudy_api/views.py

```diff
@@ -0,0 +1,165 @@
+
+from enrollment.api import add_enrollment, get_enrollment
+from enrollment.views import ApiKeyPermissionMixIn
+from lms.djangoapps.instructor.views.tools import get_student_from_identifier
+from openedx.core.lib.api.authentication import OAuth2AuthenticationAllowInactiveUser
+from openedx.core.lib.api.permissions import ApiKeyHeaderPermission
+from openedx.core.djangoapps.user_api.accounts.api import check_account_exists
+from student.models import UserProfile
+from third_party_auth.models import OAuth2ProviderConfig
+
+log = logging.getLogger(__name__)
+
+
+class CreateUserAccountWithoutPasswordView(APIView):
+    """
+    Create user account.
+    """
+    authentication_classes = (OAuth2AuthenticationAllowInactiveUser,)
+    permission_classes = (ApiKeyHeaderPermission,)
+
+    _error_dict = {
+        "username": "Username is required parameter.",
+        "email": "Email is required parameter.",
+        "uid": "Uid is required parameter."
+    }
+
+    def _check_available_required_params(self, parameter, parameter_name, values_list=None):
+        """
+        Check required parameter is correct.
+        If parameter isn't correct ValueError is raised.
+        :param parameter: object
+        :param parameter_name: string. Parameter's name
+        :param values_list: List of values
+        :return: parameter
+        """
+        if not parameter or (values_list and isinstance(values_list, list) and parameter not in values_list):
+            raise ValueError(self._error_dict[parameter_name].format(value=parameter))
+        return parameter
+
+    def post(self, request):
+        """
+        Create user by the email and the username.
+        """
+        data = dict(request.data.iteritems())
+        data['honor_code'] = "True"
+        data['terms_of_service'] = "True"
+        first_name = request.data.get('first_name', '')
+        last_name = request.data.get('last_name', '')
+
+        try:
+            email = self._check_available_required_params(request.data.get('email'), "email")
+            username = self._check_available_required_params(request.data.get('username'), "username")
+            uid = self._check_available_required_params(request.data.get('uid'), "uid")
+            if check_account_exists(username=username, email=email):
+                return Response(data={"error_message": "User already exists"}, status=status.HTTP_409_CONFLICT)
+            if UserSocialAuth.objects.filter(uid=uid).exists():
+                return Response(
+                        data={"error_message": "Parameter 'uid' isn't unique."},
+                        status=status.HTTP_409_CONFLICT
+                )
+            data['name'] = "{} {}".format(first_name, last_name).strip() if first_name or last_name else username
+            data['first_name'] = first_name
+            data['last_name'] = last_name
+            data['password'] = uuid4().hex
+            for param in ('is_active', 'allow_certificate'):
```

Comment:
do not like this construction I would prefer if you use comprehension dict something like this:
```
data.update({k: True for k in ('is_active', 'allow_certificate') if k not in data})
```

Reply:
rewrote